hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -159,6 +161,41 @@ public static HazelcastRelOptTable createRelTable(
         return new HazelcastRelOptTable(relTable);
     }
 
+    public static HazelcastRelOptTable createRelTable(
+            HazelcastRelOptTable originalRelTable,
+            HazelcastTable newHazelcastTable,
+            RelDataTypeFactory typeFactory
+    ) {
+        RelOptTableImpl newTable = RelOptTableImpl.create(
+                originalRelTable.getRelOptSchema(),
+                newHazelcastTable.getRowType(typeFactory),
+                originalRelTable.getDelegate().getQualifiedName(),
+                newHazelcastTable,
+                null
+        );
+
+        return new HazelcastRelOptTable(newTable);
+    }
+
+    public static LogicalTableScan createLogicalScanWithNewTable(
```

Comment:
I would name it simply `createLogicalScan` to make it consistent with already existing overloaded method.